snap/hooks/install

```diff
@@ -1,28 +1,42 @@
 #!/bin/bash
+# Installation hook for charmed-mongodb snap
 
 set -eux
 
+export CONF="${SNAP_DATA}/etc/mongod"
+export PBM_CONF="${SNAP_DATA}/etc/pbm/"
+export DATA="${SNAP_COMMON}/var/lib/mongodb"
+export LOG="${SNAP_COMMON}/var/log/mongodb"
+
+# If the $DATA dir is not there yet, create it
+if [ stat -c '%u' "$DATA" == 0 -o ! -d "$DATA" ]; then
+    mkdir -p $DATA
+    mkdir -p $LOG
+    chmod g+s "${LOG}"/*
+    chown -R 584788:root "${SNAP_COMMON}"/*
+    chgrp root "${LOG}"*
+else
+    mkdir -p $LOG
+    chmod g+s "$LOG"*
+    chgrp root "${LOG}"*
+fi
+
 # Create the necessary parent directories 
-mkdir -p "${SNAP_DATA}/etc/pbm/"
-mkdir -p "${SNAP_DATA}/etc/mongod/"
-mkdir -p "${SNAP_COMMON}/var/lib/mongodb/"
-mkdir -p "${SNAP_COMMON}/var/log/mongodb/"
+mkdir -p "${PBM_CONF}"
+mkdir -p "${CONF}"
 
 # Copy over the mongod.conf and create needed directories/files
-MONGO_CONFIG_FILE="${SNAP_DATA}/etc/mongod/mongod.conf"
+MONGO_CONFIG_FILE="${CONF}/mongod.conf"
```

Comment:
maybe this can also be added to the top?

Reply:
Sure